src/core/ext/filters/client_channel/client_channel.cc

```diff
@@ -1697,13 +1642,12 @@ ChannelData::ChannelData(grpc_channel_element_args* args, grpc_error** error)
   // Get default service config
   const char* service_config_json = grpc_channel_arg_get_string(
       grpc_channel_args_find(args->channel_args, GRPC_ARG_SERVICE_CONFIG));
-  if (service_config_json != nullptr) {
-    *error = GRPC_ERROR_NONE;
-    default_service_config_ = ServiceConfig::Create(service_config_json, error);
-    if (*error != GRPC_ERROR_NONE) {
-      default_service_config_.reset();
-      return;
-    }
+  if (service_config_json == nullptr) service_config_json = "{}";
```

Comment:
what happens if initially the resolver does not provide a service config, but later provides an incorrect service config?
In that case, are we to continue using the previously used "no service config" or are we to go into transient failure?

Also, please add a comment around the semantics of `default_service_config_` that no default service config provided as an arg means an empty service config being used as the default

Reply:
If the resolver returns a service config error on its first result, we go into TRANSIENT_FAILURE, and we stay there until it returns a non-error service config (i.e., either no service config or a valid service config).

If the resolver does not provide a service config, the default service config will be used, but otherwise this is considered the same as if it returned a valid config.

Once we have a valid config, if the resolver subsequently returns a service config error, we keep using the previous valid config.

I've added a comment about `default_service_config_`.